src/Microsoft.ML.Data/Transforms/RowShufflingTransformer.cs

```diff
@@ -634,7 +634,9 @@ protected override bool MoveNextCore()
                 while (_liveCount < _poolRows && !_doneConsuming)
                 {
                     // We are under capacity. Try to get some more.
-                    while (_toConsumeChannel.Reader.WaitToReadAsync().GetAwaiter().GetResult())
+                    var waitToReadAwaiter = _toConsumeChannel.Reader.WaitToReadAsync().GetAwaiter();
+
+                    while (waitToReadAwaiter.IsCompleted && waitToReadAwaiter.GetResult())
```

Comment:
Can you add a comment here stating that the check to verify the operation is complete is necessary before calling `GetResult()`?

Reply:
Dear sir, I have same error with : 

            var option = new SdcaMaximumEntropyMulticlassTrainer.Options()
            {
                FeatureColumnName = "SearchModel",
                LabelColumnName = "Label",
                NumberOfThreads = 1
            };
            var maximumEntropy = ctx.MulticlassClassification.Trainers.SdcaMaximumEntropy(option);
            var pipeline =ctx.Transforms.Conversion.MapValueToKey("Label", "BestDictMatchModelText") 
                    .Append(ctx.Transforms.Text.FeaturizeText(inputColumnName: "VEHICLE_MODEL",
                        outputColumnName: "SearchModel"))
                    .Append(maximumEntropy)
                    .Append(ctx.Transforms.Conversion.MapKeyToValue("PredictedLabel"));

            var part1 = decoratedWithFuzzyList;//.Take(1000);

            var dataReader = ctx.Data.LoadFromEnumerable(part1);
            ThisModel = pipeline.Fit(dataReader);

-------------------------------------
So, when part1 contains more than a 1000 items, Fit throws:

   at System.Threading.Channels.AsyncOperation.ThrowIncompleteOperationException()
   at System.Threading.Channels.AsyncOperation`1.GetResult(Int16 token)
   at Microsoft.ML.Transforms.RowShufflingTransformer.Cursor.MoveNextCore()
   at Microsoft.ML.Data.RootCursorBase.MoveNext()
   at Microsoft.ML.Trainers.TrainingCursorBase.MoveNext()
   at Microsoft.ML.Trainers.SdcaTrainerBase`3.TrainCore(IChannel ch, RoleMappedData data, LinearModelParameters predictor, Int32 weightSetCount)
   at Microsoft.ML.Trainers.StochasticTrainerBase`2.TrainModelCore(TrainContext context)
   at Microsoft.ML.Trainers.TrainerEstimatorBase`2.TrainTransformer(IDataView trainSet, IDataView validationSet, IPredictor initPredictor)
   at Microsoft.ML.Trainers.TrainerEstimatorBase`2.Fit(IDataView input)
   at Microsoft.ML.Data.EstimatorChain`1.Fit(IDataView input)


Can you please consider to fix this?

Reply:
I have this same issue with version 1.5.0. I can fit 1046 rows in the shuffled set. 1047 rows will give the `The asynchronous operation has not completed` exception.
Test with the daily 1.5.2 package, and now the error is gone.